jcl/src/java.management/share/classes/java/lang/management/ThreadInfo.java

```diff
@@ -755,17 +757,44 @@ public int getPriority() {
 	public String toString() {
 		// Since ThreadInfos are immutable the string value need only be
 		// calculated the one time
+		final String ls = System.lineSeparator();
 		if (TOSTRING_VALUE == null) {
 			StringBuilder result = new StringBuilder();
-			result.append(threadName + " " + threadId + " " + threadState); //$NON-NLS-1$ //$NON-NLS-2$
+			/*[IF Java11]*/
+			result.append(String.format("\"%s\" prio=%d Id=%d %s",  //$NON-NLS-1$
+					threadName, Integer.valueOf(priority), Long.valueOf(threadId), threadState));
+/*[ELSE]*/
+			result.append(String.format("\"%s\" Id=%d %s",  //$NON-NLS-1$
+					threadName, Long.valueOf(threadId), threadState));
+/*[ENDIF]*/
+			if (State.BLOCKED == threadState) {
+				result.append(String.format(" on %s owned by \"%s\" Id=%d",  //$NON-NLS-1$
+						lockName, lockOwnerName, Long.valueOf(lockOwnerId)));
+			}
+			result.append(ls);
```

Comment:
The original code only appends the ls inside the following block. Is this change intentional? I will change the output for the else case, which adds ` null` at the end.

Reply:
Thanks for catching that.